### PR TITLE
Add CSIu parser for unix

### PR DESCRIPTION
**Pull request by blaggacao**
closes #595 

This is used for [`helix-editor`](https://github.com/helix-editor/helix) keymaps.

### CODE REVIEW

File: src/event/sys/unix/parse.rs [hunk 1]
```diff
@@ -241,6 +242,53 @@ pub(crate) fn parse_csi_modifier_key_code(buffer: &[u8]) -> Result<Option<Intern
...
+    let keycode = {
+        if let Some(c) = char::from_u32(codepoint) {
+            match c {
+                '\x1B' => KeyCode::Esc.into(),
+                '\r' => KeyCode::Enter.into(),
+                // Issue #371: \n = 0xA, which is also the keycode for Ctrl+J. The only reason we get
+                // newlines as input is because the terminal converts \r into \n for us. When we
+                // enter raw mode, we disable that, so \n no longer has any meaning - it's better to
+                // use Ctrl+J. Waiting to handle it here means it gets picked up later
+                '\n' if !crate::terminal::sys::is_raw_mode_enabled() => KeyCode::Enter.into(),
```
Comment (blaggacao) [inline src/event/sys/unix/parse.rs:273]: I do not fully understand if the raw-mode check is warranted:

CSIu decimal ASCII encoding should be rather unambigous
```
10 -> LF
13 -> CR
```

**Review by blaggacao — COMMENTED** [review]
One remaining question..

File: src/event/sys/unix/parse.rs (hunk 1 above)
Comment (sigmaSd) [reply src/event/sys/unix/parse.rs:273]: This pr have the context: https://github.com/crossterm-rs/crossterm/pull/373

**Review by TimonPost — APPROVED** [review]
Late response. Anyhow, thanks! I would prefer to see some tests for this like the other codes. I can assume it work but I don't have the machine to test it myself. Anyhow can't do much harm better some support then non at all.

**Pull request merged**